### ISSUE TITLE
Can access tags in dataset

### ISSUE THREAD

**Issue by nono0481** [hiptest/hiptest-publisher#124]
Hey,

 I have rewritte the template of _scenario.hbs to generate comment with tags group before test function.

With this I can run phpunit with attribut --group to launch only tags I want :

> phpunit spec/ --group Execution:Auto --group Dev:Done

The problem is for scenario that contain a datatable.
In the template _dataset.hbs I have not access to the tags informations of scenario and I can't add comment with tag.

Code of my _scenario.hbs :
```
{{#if has_tags?}}
/**
{{#each rendered_children.tags}} * @group {{{this}}}
{{/each}} */{{/if}}
{{#if has_datasets?}}{{> item_as_function}}
{{rendered_children.datatable }}{{else}}public function test{{{ camelize_upper rendered_children.name }}}{{#if rendered_children.uid}}Uid{{{camelize_upper rendered_children.uid}}}{{/if}}()
{{#curly}}
{{> body}}
{{/curly}}{{/if}}
```

Code of my _dataset.hbs :
```
=======HERE I WANT TO ADD COMMENT WITH TAGS========
public function test{{camelize_upper scenario_name}}{{{camelize_upper rendered_children.name}}}{{#if rendered_children.test_snapshot_uid}}Uid{{{camelize_upper rendered_children.test_snapshot_uid}}}{{/if}}()
{{#curly}}
  $this->{{{ camelize_lower scenario_name }}}({{{ join rendered_children.arguments ', ' }}});
{{/curly}}
```

Is it possible to have access to tags information in dataset.hbs ?

**Comment by vincent-psarga** [comment]
Hi Nono,

This is not possible at the moment (the tags are in a different branch of the syntaxic tree than the datasets, so when rendering the dataset the tags are not in the same context).

It could be something doable that said. Two possible solutions:
 - either by changing RenderContextMaker#walk_dataset (https://github.com/hiptest/hiptest-publisher/blob/master/lib/hiptest-publisher/render_context_maker.rb) so it also gives references to the tags
 - either by adding a node modifier (https://github.com/hiptest/hiptest-publisher/tree/master/lib/hiptest-publisher/node_modifiers) in order to add the reference to the tags in the datasets

The second one seems better as it could be use the rendering system to render correctly the tags.

We'll discuss that with the team.

Best regards,
Vincent

**Comment by HoudaDabou** [comment]
Hi @nono0481,

Do you have the opportunity to try the proposed Vincent's solution?

**Comment by nono0481** [comment]
Hello,

Yes, I have some

**Comment by nono0481** [comment]
Hello @HoudaDabou,

I did not test Vincent's proposals because I was waiting for a correction from the team. I do not have enough skill to edit and test the core of hiptest-publisher.

However I can provide you the template files I use, if you want.

**Comment by HoudaDabou** [comment]
yes, please forward me this template :)

**Comment by nono0481** [comment]
Hi,

I give you my templates files :
**_scenario.hbs :**
```
{{#if has_tags?}}
/**
{{#each rendered_children.tags}} * @group {{{this}}}
{{/each}} */{{/if}}
{{#if has_datasets?}}{{> item_as_function}}
{{rendered_children.datatable }}{{else}}public function test{{{ camelize_upper rendered_children.name }}}{{#if rendered_children.uid}}Uid{{{camelize_upper rendered_children.uid}}}{{/if}}()
{{#curly}}
{{> body}}
{{/curly}}{{/if}}
```

**dataset.hbs**
```
public function test{{camelize_upper scenario_name}}{{{camelize_upper rendered_children.name}}}{{#if rendered_children.test_snapshot_uid}}Uid{{{camelize_upper rendered_children.test_snapshot_uid}}}{{/if}}()
{{#curly}}
  $this->{{{ camelize_lower scenario_name }}}({{{ join rendered_children.arguments ', ' }}});
{{/curly}}
```
I would like to add same scenario comments in each function in dataset.

**Comment by HoudaDabou** [comment]
Thank you!

We'll get back to as soon as discuss with team :)

**Comment by nono0481** [comment]
Hi, 
do you have any feebback from the team ? Do you have a release date for this evolution ?

**Comment by HoudaDabou** [comment]
Hi Nono!

Unfortunately not yet, just give me few time by the end of the week, I'll test your template and come back to you =)

Thanks for your patience.

**Comment by nono0481** [comment]
Hi @HoudaDabou,

I'm sorry to bother you again but did you have time to perform the evolution?

**Comment by HoudaDabou** [comment]
Hi @nono0481!

I haven't forgotten you but I have some difficulties to apply the above instructions, moreover Vincent is on leave at the moment...

Another of our experts, Christophe B., is willing to try, but he will only be able to look into it from Monday, May 20. 

Feel free to contact us from the in-chat available from your HipTest account ;)

Kind regards,

**Comment by nono0481** [comment]
Hi @HoudaDabou,

I'm sorry to bother you again but did you have time to perform the evolution?

(I have not find the in-chat from my HipTest account that's why I retry in this issue !)

**Comment by HoudaDabou** [comment]
Hey @nono0481!

My apologies! we had a lot of work before the summer and now some ones of my colleagues who can help you are on vacation :(

At this time, I can just suggest to review the changes that you'll made to have the expected result. So, the first proposal made by Vincent P. is not the best but it's the easiest way in my opinion. 
Thanks to start to modify the [RenderContextMaker#walk_dataset](https://github.com/hiptest/hiptest-publisher/blob/master/lib/hiptest-publisher/render_context_maker.rb) to get references to the tags, then submit it to us to review it.

_P.S.: pay attention to change the right file, so locate the path - in my case is: /home/houda/.rvm/gems/ruby-2.5.3/gems/hiptest-publisher-1.25.0/lib/hiptest-publisher/render_context_maker.rb - then make the changes in this one._

**Comment by nono0481** [comment]
Hey @HoudaDabou ,

 I create a Pull Request to update "render_context_marker.rb" file and add two properties in walk_dataset function : has_tags and tags

I use they in "dataset.hbs" file to print tags comment before the method name :
```
{{#if has_tags?}}/**
{{#each tags}} * @group {{{this}}}
{{/each}} */{{/if}}
public function test{{camelize_upper scenario_name}}{{{camelize_upper rendered_children.name}}}{{#if rendered_children.test_snapshot_uid}}Uid{{{camelize_upper rendered_children.test_snapshot_uid}}}{{/if}}()
{{#curly}}
  $this->{{{ camelize_lower scenario_name }}}({{{ join rendered_children.arguments ', ' }}});
{{/curly}}
```
And finally it works fine !

Can you have a look on my pull request #142 ?

**Comment by HoudaDabou** [comment]
Hello Nono!

I tried too and your changes work very well, good work =)

**Comment by vincent-psarga** [comment]
Closing as #142 has been merged and released in 1.27.1
And thanks again for the contribution :)
